spec/System/HapistranoSpec.hs

```diff
@@ -146,7 +147,18 @@ justExec path cmd' =
 -- | Run 'Hapistrano' monad locally.
 
 runHap :: Hapistrano a -> IO a
-runHap = Hap.runHapistrano Nothing
+runHap m = do
+  let printFnc dest str =
+        case dest of
+          StdoutDest -> putStr str
+          StderrDest -> hPutStr stderr str
+  r <- Hap.runHapistrano Nothing printFnc m
+  case r of
+    Left n -> do
+      expectationFailure ("Failed with status code: " ++ show n)
+      return undefined
```

Comment:
This is necessary because `expectationFailure` has incorrect type. It should return `IO a`, but it returns `IO ()` instead. This fixed unit type prevents from using it in the same `case` expression with with more general type. The reason `return undefined` is OK and `IO ()` is not OK is that `expectationFailure` always throws so control flow will never get to `return undefined`. It could be just `undefined` or anything at all with general enough type.